Creating report from TG functuion -> done!

algmironov opened the pull request:
По пути -> Главное меню -> Отправить отчет Теперь можно отправить текст, фото и фото с текстом.

algmironov committed:
Реализована отправка текста и фото для отчета -> создание нового отчета родителем в боте

algmironov committed:
Реализована отправка текста и фото для отчета -> создание нового отчета родителем в боте
UPD: -> реализована возможность использовать подпись к фото как текст отчета

Delegant commented on src/main/java/pro/sky/telegrambot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -389,8 +434,91 @@ public void handleUserMessages(Function<String, Boolean> whatIsMenu,
+            doSendMessage.accept("DEFAULT_MENU_TEXT", "MAIN_MENU");
+        } else {
+            doSendMessage.accept("ERROR_COMMAND_TEXT", "CALL_VOLUNTEER_MENU");
+        }
```
Очень похож метод на метод юзера, надо бы как то это оптимизировать в перспекииве

Delegant reviewed (approved):
Немало ты сегодня добавил 👍

algmironov committed:
Добавлен просмотр отчета по нажатию на него (из "Просмотреть отчеты" -> список отчетов) -> отправляются все фото отчета + текст.
В текст кнопок отчета теперь добавлено количество фото
Добавлено меню для операций с отчетом
Добавлено новое свойство модели Report -> ReadStatus (прочитано/не прочитано/ требует доработок)

algmironov committed:
Доделана логика проверки отчетов, отправки уведомлений родителям о необходимости обновить отчет (текст или фото) или об одобрении отчета волонтером.
//todo -> доделать логику реакции родителя на уведомления о необходимости обновить отчет.

algmironov committed:
Обновил логику отправки фото с локацией
- изменил Consumer на TriConsumer
- добавил отдельные методы для формирования отправки фото в меню сервисе (на случай, если потребуется использовать оригинальный метод отправки фото

algmironov committed:
Логика работы отправки уведомления о необхожимости добавить фото/текст + обработка команд юзером + просмотр своего испытательного периода юзером + (частично) просмотр своих отчетов
